### ISSUE TITLE
Test detection error

### ISSUE THREAD

**Issue by gjm441** [karta0807913/caffe-yolo-9000#7]
I have trained the model as instructions, and run the code for testing VOC image, but it has poor result.
After I learn the darknet yolo v2 source code, I find that the box_data_layer may be wrong, because it still takes side_*side_*(1+1+1+4), the same with yolo v1, that means one cell still only have one label class.

I have modified the box_data_layer implementation according to yolo v2 based on darknet. But it has some link problems.

**Comment by karta0807913** [comment]
oops, you're right!
I will try to fix that
The reason I did not change before is that I might to rewrite yolo v1
maybe I can add a new layer?

**Comment by zhengstake** [comment]
I have just started looking at this code base. I'd want to make sure that it is got the basic YOLO V2 function implemented correctly. I am a little bit confused by this issue and the answer.

I thought the box_data_layer was just importing the ground truth labels and output a top_blob[1] = 13*13*(1+1+1+4), which for each cell contains 'difficult, isobj flag, class label and 4 box parameters. This should have nothing to do with the per anchor box predictions. This ground truth label blob also seems to be used consistently in the yolo_v2_loss layer.

So what exactly is the expected changes to be made for YOLO V2?

**Comment by karta0807913** [comment]
I just change the loss function base [this](https://github.com/yeahkun/caffe-yolo) implement
so that part is yolov1 and i didn't change that
